src/record-error-archival/src/DatabaseClient.ts

```diff
@@ -0,0 +1,129 @@
+import { Client } from "pg"
+import { logger } from "shared"
+
+export type ArchivedErrorRecord = {
+  messageId: string
+  errorId: number
+  archivedAt: Date
+  archivedBy: string
+  archiveLogId: number
+}
+
+// Produces the string "$1, $2, $3..." for the given range
+const wherePlaceholderForRange = (range: number) => [...Array(range).keys()].map((x) => `$${x + 1}`).join(", ")
+
+export default class DatabaseClient {
+  private host: string
+
+  private user: string
+
+  private password: string
+
+  private database: string
+
+  private useSsl: boolean
+
+  private schema: string
+
+  private postgres: Client
+
+  private archiveGroupLimit: number
+
+  constructor(
+    host: string,
+    user: string,
+    password: string,
+    useSsl: boolean,
+    database: string,
+    schema: string,
+    archiveGroupLimit: number
```

Comment:
There's a nice shorthand for this type of initialisation of private constructors:

https://dev.to/satansdeer/typescript-constructor-shorthand-3ibd